### PR TITLE
feat: generate interactive base library import examples via `mo-doc`

### PR DESCRIPTION
**Pull request by rvanasa**
This PR sets the groundwork for generating interactive examples in the base library documentation. 

**Progress:**
- [x] Add a `--name <string>` flag to `mo-doc` to enable examples using the given Vessel package name
- [x] Include an interactive `import Module "mo:name/...Module";` code snippet at the top of each generated Markdown file. The goal here is to give developers a sandbox in case an example is missing (and to provide a quick copy/paste import snippet)

### CODE REVIEW

**Comment by crusso** [comment]
If you name the import fragments, then the function examples can include them in the code block specifier. See the doc/Readme.md for the syntax.

**Comment by rvanasa** [comment]
Great idea; thanks!

**Comment by rvanasa** [comment]
This should be ready to merge as a preparatory step for mass-producing base library examples.

File: doc/md/base/Array.md [hunk 1]
```diff
@@ -1,205 +1,486 @@
...
+Provides extended utility functions on Arrays. Note the difference between
+mutable and non-mutable arrays below.
+
+WARNING: If you are looking for a list that can grow and shrink in size,
+it is recommended you use either the Buffer class or the List class for
+those purposes. Arrays must be created with a fixed size.
+
+Import from the base library to use this module.
+```motoko name=import
+import Array "mo:base/Array";
```
[the remark this answers is not in the document]
Comment (rvanasa) [reply doc/md/base/Array.md:16]: It turns out the only other Markdown file in our documentation with `name=import` is following the same convention:
https://github.com/dfinity/motoko/blob/master/doc/md/mutable-state.md?plain=1#L145-L147

This behavior is also opt-in via the `--name` flag, so it won't break anyone's existing docs.

**Pull request closed without merging**